Insights: add dashboard specification (#4367)
mozilla/pontoon: specs/0123-insights-dashboard.md

- Feature Name: Community health score (CHS) panel
- Created: 2026-08-04
- Associated PR: [#4189](https://github.com/mozilla/pontoon/pull/4189)

# Summary

Introduce an admin-only Community health score panel within the global Insights dashboard that provides a monthly breakdown of the Community health score and its underlying metrics, along with historical score tracking. Also make the historical tracking available on locale Insights dashboards.

# Motivation

Currently, the Community health score is calculated via a series of moz-l10n scripts and Google Sheets formulas, collected manually on a monthly basis. Other than the tedious nature of needing to do this manually, PMs need better options to analyze CHS data directly from Pontoon.

# Feature explanation

## Community health score aggregation

On the first day of each month, a scheduled job collects telemetry on the state of each locale for the **previous month**. The data collected for **each locale** is compiled into the following metrics:

> - `active_managers`: Managers whose combined total of performed reviews and approved translations during the past 12 months exceeds 500.

Reviewer: Now I realized most of these are configurable. Please change to something like this:

```suggestion
- `active_managers`: Managers whose combined total of performed reviews and approved translations during the past 12 months exceeds `MANAGER_STRING_THRESHOLD` (default: 500).
```

- `active_translators`: Translators whose combined total of performed reviews and approved translations during the past 12 months exceeds 400.
- `active_contributors`: Contributors whose approved translations during the past 12 months meets or exceeds 200.
- `all_contributors`: Contributors whose combined total of approved translations, rejected translations and pending suggestions during the past 12 months meets or exceeds 200.
- `new_signups`: New signups whose approved translations during the past 12 months meets or exceeds 100.
- `key_projects_enabled`: number of projects that are classified as key projects via the `is_chs_project` flag

> - `completion`: locale completion, as a percentage of (**approved strings** + **strings with warnings**) out of **total strings**

Reviewer: ```suggestion
- `key_projects_enabled`: Number of projects that are classified as key projects via the `is_chs_project` flag.
- `completion`: Locale completion, expressed as a percentage of **approved strings** + **strings with warnings** out of **total strings**.
```

> The data used is then computed into the following scores based on predetermined thresholds. Finally the scores are summated to a **Community health score** for that locale on that month which is used for measuring locale health.

Reviewer: What following scores and what thresholds?

> The data used is then computed into scores based on predetermined metric thresholds, which are set as environment variables for admin customization. Finally the scores are summated to a **Community health score** for that locale on that month which is used for measuring locale health.

Reviewer: Imagine you are a dev, presented with this spec. This information is insufficient for implementation of the computation of the scores. We should list the environment variables in question and describe the formula / algorithm for the calculation of the score.

## Community health score panel

The new **Community health score** panel located at `/insights` consists of a widget, a table of locales and a graph.

The widget at the top of the panel contains **locale search** and buttons titled `Edit Locales` and `Show scores`. `Edit Locales` swaps the panel contents for a selector that enables admins to change locale availability on the CHS panel. `Show scores` toggles default and score view of components that make up the CHS.

The table of locales represents the locales the admin wishes to observe based on the most recent month's CHS data. Each table row contains the following information:
- **Locale code** linking to individual locale pages
- Number of **active managers**
- Number of **active translators**
- Number of **active contributors**
- Number of **all contributors**
- Number of **new signups**

> - Number of **enabled key projects** (%)

Reviewer: ```suggestion
- Number of **enabled key projects**
```

- Overall locale **completion** (%)
- CHS

> Beside each statistic, an up/down arrow combined with a number represents the delta over the previous month's statistic. Data that has no change is not represented with a delta. For counts (contributor, translator, manager counts etc.), each value is represented as an integer. For the scores and percentages, each value is represented with 2 decimal places.

Reviewer: ```suggestion
Beside each statistic, an up or down arrow and a number indicate the change from the previous month. If there is no change, no delta is shown. Counts (e.g., contributors, translators, and managers) are displayed as integers, while scores and percentages are displayed to two decimal places.
```

> - Feature Name: Insights Dashboard

Reviewer: ```suggestion
- Feature Name: Community health score (CHS) panel
```

- Created: 2026-08-04
- Associated PR: [#4189](https://github.com/mozilla/pontoon/pull/4189)

# Summary

> Introduce an admin-only dashboard within `/insights` where PMs can observe the monthly changes to the Community Health Score through a deconstruction of its algorithm, and implement additional statistics tracking within global and locale Insights pages for PM ingestion.

Reviewer: ```suggestion
Introduce an admin-only Community health score panel within the global Insights dashboard that provides a monthly breakdown of the Community health score and its underlying metrics, along with historical score tracking. Also make the historical tracking available on locale Insights dashboards.
```

# Motivation

> Currently, the CHS is calculated via a series of moz-l10n scripts and Google Sheets formulas, collected manually on a monthly basis. Other than the tedious nature of needing to do this monthly, PMs need better options to analyze CHS data holistically.

Reviewer: ```suggestion
Currently, the Community health score is calculated via a series of moz-l10n scripts and Google Sheets formulas, collected manually on a monthly basis. Other than the tedious nature of needing to do this manually, PMs need better options to analyze CHS data holistically.
```

# Feature explanation

> ## Community health score activity

Reviewer: ```suggestion
## Community health score panel
```

> This feature located at `/insights` consists of a table of locales similar to the `/teams` page, and a multi-line graph.

Reviewer: ```suggestion
This new "Community health score" panel located at `/insights` consists of a table of locales similar to the `/teams` page, and a graph.
```

> - Top menu title: `Community health activity`

Reviewer: ```suggestion
```

> - Dashboard:

Reviewer: ```suggestion
- Table:
```

Reviewer: I would make "Table" and "Graph" subsections and only use bullet points where needed.

>   - Widget: contains locale search and buttons titled `Configuration` and `Show Scores`

Reviewer: There's no button called "Configuration".

Also, why is that line a bullet point? Same question for other lines in the same nesting level?

Author: Button no longer exists yes, I am using [0118-messaging-center.md](github.com/mozilla/pontoon/blob/main/specs/0118-messaging-center.md) as a template which is why there is nested levels.

>     - `Configuration`: links to a separate page that enables admins to change locale availability on the dashboard and CHS activity graph

Reviewer: It's not a separate page.

>     - `Show Scores`: toggles default and score view of components that make up the CHS

Reviewer: How are the score values calculated?

Author: added a section on notifying how scores are calculated.

>   - Layout: tabular

Reviewer: ```suggestion
```

  - Each row contains the following information:
    - Locale code linking to individual locale pages

>     - \# of managers with > 500 submissions trailing 12 months / 1

Reviewer: ```suggestion
    - Number of managers with > 500 submissions trailing 12 months / 1
```

    - \# of translators with > 400 submissions trailing 12 months / 2
    - \# of contributors with > 200 submissions trailing 12 months / 2
    - \# of contributors (all) this month / 2
    - \# of new signups with > 100 submissions / 2

>     - \# of enabled projects / total # of key projects * 100%

Reviewer: ```suggestion
    - Number of enabled key projects / total number of key projects * 100%
```

>     - \# of enabled projects / total # of key projects * 100%

Reviewer: Why #?

What does trailing mean?

Why divide by 1?

Are these absolute numbers (default values) or scores?

Author: > Why #?

"#" represents "Number of" -> I'll change it to the word format
> What does trailing mean?

"Trailing 12 months" refers to the most recent 12 months -> I'll change it to "most recent 12 months"
> Why divide by 1

This is a good point. The division by number refers to the minimum threshold required for full points. Need to redefine this.
> Are these absolute numbers (default values) or scores?

These are supposed to be absolute numbers

Author: I deleted this section in favor for a new section explaining how contributors are counted.

    - Overall completion percentage of locale

>     - CHS, written as a float

Reviewer: How is CHS Calculated?

Author: Added a section to briefly explain how CHS is calculated, although this should definitely be expounded upon via the external CHS docs which have yet to be created.

  - Beside each statistic, an up/down arrow combined with a smaller number, indicating increase or decrease, presents the delta over the previous month's statistic. Data that has no change is not represented with a delta.
  - Value formatting:
    - For counts (contributor, translator, manager counts etc.), each value is an integer without exception.

>     - For the scores and percentages, each value is fixed at 2 places (e.g. `20.00`) without exception.

Reviewer: What does "without exception" mean?

Reviewer: Also, these lines should not be bullet points.

- Graph:
  - Tooltip: describes basic functionality, which is to track CHS per locale
  - Legend: contains locales and `Average (all locales)`, which can be selected/deselected to display corresponding CHS datapoints over time.

>   - Display: displays CHS scores for selected locales trailing 12 months. As soon as the CHS calculation automation on Pontoon runs on the first of each month, the previous month's scores will be automatically available for display.

Reviewer: These should be written as paragraphs, not bullet points.

## Insights Configuration

The dashboard page has a button named `Edit Locales`. Upon clicking it, admins are redirected to a locale selector that enables them to add/remove locales from their dashboard view.

## Locale Insights changes

> The `/<locale-code>/insights/` page currently includes statistics and useful information regarding locale health and performance. The main change includes a new graph as follows:

Reviewer: Please reference https://github.com/mozilla/pontoon/blob/main/specs/0108-community-health-dashboard.md

- Feature Name: Community health score (CHS) panel
- Created: 2026-08-04
- Associated PR: [#4189](https://github.com/mozilla/pontoon/pull/4189)

# Summary

Introduce an admin-only Community health score panel within the global Insights dashboard that provides a monthly breakdown of the Community health score and its underlying metrics, along with historical score tracking. Also make the historical tracking available on locale Insights dashboards.

# Motivation

Currently, the Community health score is calculated via a series of moz-l10n scripts and Google Sheets formulas, collected manually on a monthly basis. Other than the tedious nature of needing to do this manually, PMs need better options to analyze CHS data directly from Pontoon.

# Feature explanation

## Community health score aggregation

> On the first of each month, an automation is run that collects telemetry on the state of each locale for the **previous month**. The data collected for **each locale** is compiled into the following datapoints:

Reviewer: ```suggestion
On the first day of each month, a scheduled job collects telemetry on the state of each locale for the **previous month**. The data collected for **each locale** is compiled into the following metrics:
```

> - `active_managers`: managers with **reviews performed** + **approved translations** > 500 in the past 12 months

Reviewer: I'd raphrase these to make them more readable, e.g.:

```suggestion
- `active_managers`: Managers whose combined total of performed reviews and approved translations during the past 12 months exceeds 500.
```

- `active_translators`: translators with **reviews performed** + **approved translations** > 400 in the past 12 months

> - `active_contributors`: contributors with **approved translations** >= 200 in the past 12 months

Reviewer: Is this threashold really >= (i.e. "at least"), while the ones above are > ("more than")?

Author: Yes.

- `all_contributors`: contributors with **approved translations** + **rejected translations** + **pending suggestions** >= 200 in the past 12 months
- `new_signups`: new signups with **approved translations** >= 100 in the past 12 months
- `key_projects_enabled`: number of projects that are classified as key projects via the `is_chs_project` flag
- `completion`: locale completion, as a percentage of (**approved_strings** + **strings_with_warnings**) out of **total strings**

The data used is then computed into the following scores based on predetermined thresholds:

- `active_managers_score`: score representation of `active_managers`
- `active_translators_score` score representation of `active_translators`
- `active_contributors_score` score representation of `active_contributors`
- `all_contributors_score` score representation of `all_contributors`
- `new_signups_score` score representation of `new_signups`
- `key_projects_enabled_score` score representation of `key_projects_enabled`

> - `completion_score` score representation of `completion`

Reviewer: This paragraphs doesn't add much value. Instead, we should explain the formula for calculating each of these scores. 

Think of the spec as the containing sufficient information to develop the feature. It will also be helpul in the future when we'll need to understand how the score works, so we don't need to decypher the code.

> Finally the scores are combined into a **Community health score** for that locale on that month which is used for measuring locale health.

Reviewer: How excatly (mathematically) are they combined?

The widget at the top of the panel contains **locale search** and buttons titled `Edit Locales` and `Show scores`. `Edit Locales` swaps the panel contents for a selector that enables admins to change locale availability on the CHS panel. `Show scores` toggles default and score view of components that make up the CHS.

The table of locales represents the locales the admin wishes to observe based on the most recent month's CHS data. Each table row contains the following information:
- **Locale code** linking to individual locale pages
- Number of **active managers**
- Number of **active translators**
- Number of **active contributors**
- Number of **all contributors**
- Number of **new signups**
- Number of **enabled key projects** (%)
- Overall locale **completion** (%)
- CHS

Beside each statistic, an up/down arrow combined with a number represents the delta over the previous month's statistic. Data that has no change is not represented with a delta. For counts (contributor, translator, manager counts etc.), each value is represented as an integer. For the scores and percentages, each value is represented with 2 decimal places.

> The graph consists of a tooltip, legend and display. The tooltip describes basic functionality. The legend contains locales and `Average (all locales)`, which can be selected/deselected to display corresponding CHS datapoints over time. The display shows CHS scores for selected locales in the past 12 months.

Reviewer: ```suggestion
The chart is a line chart that displays Community Health Score (CHS) values for the selected locales over the past 12 months. Hovering over any data point displays a tooltip with the exact CHS value for that month for all selected locales. The legend lists individual locales along with `Average (all locales)` and allows each line to be shown or hidden.
```

## Locale Insights changes

The [locale insights](https://github.com/mozilla/pontoon/blob/main/specs/0108-community-health-dashboard.md) page currently includes statistics and useful information regarding locale health and performance.

> The main change introduces a new graph called `Community health score` which contains a tooltip that describes basic functionality, which is to show the CHS of the current locale at a particular point in time. It displays CHS scores for current locale trailing 12 months. As soon as the CHS calculation automation on Pontoon runs on the first of each month, the previous month's scores will be automatically available for display.

Reviewer: ```suggestion
A new graph called `Community health score` is added to the [locale insights page](https://github.com/mozilla/pontoon/blob/main/specs/0108-community-health-dashboard.md), which shows the CHS values for the locale over the past 12 months.
```
